src/wp-includes/kses.php

```diff
@@ -1686,7 +1686,27 @@ function wp_kses_check_attr_val( $value, $vless, $checkname, $checkvalue ) {
  * @return string Filtered content.
  */
 function wp_kses_bad_protocol( $string, $allowed_protocols ) {
-	$string     = wp_kses_no_null( $string );
+	$string = wp_kses_no_null( $string );
+
+	// Short-circuit if the string starts with `https://` or `http://`. Most common cases.
+	if ( in_array( 'https', $allowed_protocols, true ) ) {
+		// Check if the string starts with `https` and it is lower case.
+		if ( 0 === strpos( $string, 'https://' ) ) {
```

Comment:
Can we use `str_starts_with` here.

Reply:
Added in https://github.com/WordPress/wordpress-develop/commit/1046682e4f2453c55f836b3fe69f529daea0babc. It is cleaner and easier to read.

Reply:
I see. It still would be a tiny bit slower on PHP 7.4 though. Also the code here doesn't need the conditional in `str_starts_with()` as the needle is hard-coded and not empty.

Reply:
Thanks @spacedmonkey for pointing.

I ran the analysis in PHP 7.4 and 8.0 and the different is 1-2%.

- 7.4
  - `strpos` ~31%
  - `str_starts_with` ~30%
- PHP 8+
  -  `strpos` ~32.5%
  - `str_starts_with` ~33.5%

As it improves performance now, improves performance further as more sites move to PHP 8+, it's easier to read, and Core has been changing existing instances of `0 === strpos()` to `str_starts_with()`, I'm agreeing to use `str_starts_with()`.